pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -383,9 +385,33 @@ func isConfiguredToOwnCloudController(cm *corev1.ConfigMap) (bool, error) {
 	return len(cloudProvider) != 1 || (cloudProvider[0] != "external" && cloudProvider[0] != ""), nil
 }
 
-func manageKubeControllerManagerConfig(ctx context.Context, client corev1client.ConfigMapsGetter, recorder events.Recorder, operatorSpec *operatorv1.StaticPodOperatorSpec) (*corev1.ConfigMap, bool, error) {
+func manageKubeControllerManagerConfig(ctx context.Context, client corev1client.ConfigMapsGetter, infrastuctureLister configv1listers.InfrastructureLister, recorder events.Recorder, operatorSpec *operatorv1.StaticPodOperatorSpec) (*corev1.ConfigMap, bool, error) {
+	cluster, err := infrastuctureLister.Get("cluster")
+	if err != nil {
+		return nil, false, err
+	}
+	topology := cluster.Status.ControlPlaneTopology
+	var config = configv1.LeaderElection{}
+	if topology == configv1.SingleReplicaTopologyMode {
+		config = leaderelection.LeaderElectionSNOConfig(config)
```

Comment:
SNO is up for a discussion, but we shouldn't change the leader election for normal clusters, since this would significantly hamper availability of KCM when a master node would go down.